.github/workflows/pull_request.yaml

```diff
@@ -13,43 +13,51 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
     - name: Cancel Previous Runs
-      uses: styfle/cancel-workflow-action@0.11.0
+      uses: styfle/cancel-workflow-action@0.12.1
       with:
         access_token: ${{ github.token }}
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v5
       with:
         fetch-depth: 0
     - name: Set up Python
-      uses: actions/setup-python@v4
+      uses: actions/setup-python@v6
       with:
         python-version: 3.8
-    - uses: google-github-actions/setup-gcloud@a48b55b3b0eeaf77b6e1384aab737fbefe2085ac
+
+    - name: Gcloud login
+      if: ${{ env.has_auth }}
+      uses: google-github-actions/auth@v2
       with:
-        version: '290.0.1'
         project_id: ${{ secrets.GCP_PROJECT_ID }}
-        service_account_key: ${{ secrets.GCP_SA_KEY }}
-        export_default_credentials: true
-      name: Gcloud Login
+        credentials_json: ${{ secrets.GCP_SA_KEY }}
+        create_credentials_file: true
+      env:
+        has_auth: ${{ secrets.GCP_PROJECT_ID && secrets.GCP_SA_KEY && 1 }}
```

Comment:
In order to not try to auth when you don't have credentials, you have to bend over backwards because GitHub doesn't give you access to `secrets` in `if`